docs/charmsindetail.md

```diff
@@ -0,0 +1,304 @@
+# Charms in detail
+## Table of contents
+
+[Best practices](##Best%20practice)
+
+[Charm writing in detail](##Charm%20Writing%20in%20detail)
+
+[New to charms?](##New%20to%20charms?)
+
+[Charm Writing](#Charm%20writing)
+
+[Example Charms](##Example%20Charms)
+
+## Best practices
+
+### Repoository Naming
+
+Naming follows the convention of charm-<charm_name> for example `charm-percona-cluster`.
+
+The rationale behind this is quick and easy searching of the 1000s of github projects along with other reasons.
+
+
+## Charm Writing in detail
+
+## New to charms?
+
+> This covers aspects of charming which will be useful when creating them
+
+### Why charms?
+
+The objective of charms is the following: take the complication of managing 100s potentially 1000s of configuration files and abstract them as python modules. Thus by abstraction creating a true insfrastructure as code.
+
+But charms are not just limited to setting up configuration, they also handle the installation of charms, and can implement actions thus providing not just configuration, but day zero to day two operations support.
+
+One of the ways that charms achieves this goal is through what are called relations, you can think about this much like the Database relation charts you might have seen in previous roles, or at university, the concept is much the same.
+
+Here's an example from the Juju GUI:
+
+![Openstack relations layout](./diagrams/juju_gui.jpg)
+
+What you are looking at here is a Kubernetes, rendered, using Juju and charms. Each of the circles represents an application, each of the lines represents what we call a relation.
+
+### How do relations work?
+
+Relations between charms at a high level work through what are called interfaces, interfaces are the glue that sticks two or more charms together and allows them to communicate.
+
+#### Some history
+
+In the old framework (reactive) charms were found on github, and used the naming convention of `interface-<interface-name>`. This stays with the new framework.
```

Comment:
I wouldn't talk about the history in a practical tutorial. Many readers of this document won't have experience with reactive anyway.